OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -120,6 +123,13 @@ public void Tick(Actor self)
 					frozenActor.TooltipInfo = tooltip.Value.TooltipInfo;
 					frozenActor.TooltipOwner = tooltip.Value.Owner;
 				}
+
+				if (ownerHasChanged[player])
+				{
+					var mod = self.TraitsImplementing<IRadarColorModifier>().FirstOrDefault();
+					frozenActor.RadarColor = mod != null ? mod.RadarColorOverride(self) : self.Owner.Color.RGB;
+					ownerHasChanged[player] = false;
+				}
```

Comment:
Owner changes aren't the only things that can change the color.  Cloaking is another specific example, and there are arbitrary other things that may also do it.